discussions/api.py

```diff
@@ -112,3 +120,42 @@ def update_discussion_user(discussion_user):
         raise DiscussionUserSyncException(
             "Error updating discussion user, got status_code {}".format(result.status_code)
         )
+
+
+def add_channel(original_search, title, name, public_description, channel_type):
+    """
+    Add the channel and associated query
+
+    Args:
+        original_search (Search):
+            The original search, which contains all back end filtering but no filtering specific to mail
+            or for percolated queries.
+        title (str): Title of the channel
+        name (str): Name of the channel
+        public_description (str): Description for the channel
+        channel_type (str): Whether the channel is public or private
+
+    Returns:
+        Channel: A new channel object
+    """
+    client = get_staff_client()
+    response = client.channels.create(
+        title=title,
+        name=name,
+        public_description=public_description,
+        channel_type=channel_type,
+    )
+    if not response.ok:
+        raise ChannelCreationException("Error creating channel: {}".format(response.content))
```

Comment:
My only feedback is that this line results in a 500 error from the API (e.g. put a space in the channel name for an invalid channel name), I suppose since this is an administrative API that we can be ok with this because it won't get used frequently, but just wanted to mention since i ran into it accidentally.

Reply:
Yeah, I didn't want to overcomplicate the validation at this point for a use case that wasn't common